crates/polars-ops/src/frame/join/iejoin/mod.rs

```diff
@@ -364,6 +364,15 @@ unsafe fn materialize_join(
     suffix: Option<PlSmallStr>,
 ) -> PolarsResult<DataFrame> {
     try_raise_keyboard_interrupt();
+    let mut sorted_left_row_idx = left_row_idx.sort(false).into_no_null_iter().peekable();
+    let left_idx_not_in_inner = (0..left.height() as u64).into_iter().filter(|left| {
```

Comment:
Indexes might be u64 or u32 depending on the `bigidx` feature:
```suggestion
    let left_idx_not_in_inner = (0..left.height() as IdxSize).into_iter().filter(|left| {
```